splash/lua_modules/wraputils.lua

```diff
@@ -109,6 +109,31 @@ local function sets_callback(func, storage)
   end
 end
 
+local function pack_callback_return_value(func)
+  return function(cb, ...)
+    local mcb = cb
+
+    if type(cb) == 'function' then
+      mcb = function(...)
+        return { cb(...) }
+      end
+    end
+
+    return func(mcb, ...)
+  end
+end
```

Comment:
Why is it necessary to pack & unpack it here? If we're returning an error, it's a single value and shouldn't be wrapped, otherwise the values originate from user-specified code chunk, shouldn't they pass through our executor intact?